src/pagerduty/token_auth_handler.bal

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+import ballerina/http;
+import ballerina/log;
+
+# Representation of the Token Auth header handler for outbound HTTP traffic.
+#
+# + authProvider - The `OutboundAuthProvider` instance
+public type TokenAuthHandler object {
+
+     *http:OutboundAuthHandler;
+
+     public auth:OutboundAuthProvider authProvider;
+
+      public function __init(auth:OutboundAuthProvider authProvider) {
+          self.authProvider = authProvider;
+      }
+
+      # Prepares the request with the Token Auth header.
+      #
+      # + req - The`Request` instance
+      # + return - Returns the updated `Request` instance or the `AuthenticationError` in case of an error
+      public function prepare(http:Request req) returns http:Request|http:AuthenticationError {
+          auth:OutboundAuthProvider authProvider = self.authProvider;
+          string|auth:Error token = authProvider.generateToken();
+          if (token is string) {
+              req.setHeader(http:AUTH_HEADER, AUTH_SCHEME_TOKEN + " token=" + token);
+              return req;
+          } else {
+              return prepareAuthenticationError("Failed to prepare request at bearer auth handler.", token);
+          }
+      }
+
+    # Inspects the request and response and calls the Auth provider for inspection.
+    #
+    # + req - The `http:Request` instance
+    # + resp - The `http:Response` instance
+    # + return - The updated `http:Request` instance, an `http:AuthenticationError` in case of an error,
+    #                 or else `()` if nothing is to be returned
+    public function inspect(http:Request req, http:Response resp) returns http:Request|http:AuthenticationError? {
+        auth:OutboundAuthProvider authProvider = self.authProvider;
+        map<anydata> headerMap = createResponseHeaderMap(resp);
+        string|auth:Error? token = authProvider.inspect(headerMap);
+        if (token is string) {
+            req.setHeader(http:AUTH_HEADER, AUTH_SCHEME_TOKEN + token);
```

Comment:
If the header is `Authorization: Token token=<token>` pattern, this should be as follows:
```suggestion
            req.setHeader(http:AUTH_HEADER, AUTH_SCHEME_TOKEN + " token=" + token);
```